### PR TITLE
Add support for RSA-PSS

### PR DESCRIPTION
**Pull request by gowthamsk-arm**
Signed-off-by: Gowtham Suresh Kumar <gowtham.sureshkumar@arm.com>

### CODE REVIEW

File: src/subcommands/create_csr.rs [hunk 1]
```diff
@@ -181,11 +181,18 @@ impl CreateCsr {
                     // Key policy specifies raw RSA signatures. RCGEN will always hash-and-sign, so fail.
                     error!("Signing key specifies raw signing only, which is not supported for certificate requests.");
                     Err(ToolErrorKind::NotSupported.into())
-                }
-                AsymmetricSignature::RsaPss { .. } => {
-                    error!("Signing key specifies RSA PSS scheme, which is not supported for certificate requests.");
```
Comment (tgonzalezorlandoarm) [inline src/subcommands/create_csr.rs:186]: Why was this not being supported for certificate requests before?

File: src/subcommands/create_rsa_key.rs [hunk 2]
```diff
@@ -27,6 +27,11 @@ pub struct CreateRsaKey {
     #[structopt(short = 's', long = "for-signing")]
     is_for_signing: bool,
 
+    /// This command creates RSA encryption keys by default. Supply this flag to create a signing key instead.
+    /// Signing keys, by default, will specify the SHA-256 hash algorithm and use PKCS#1 v1.5.
```
Comment (tgonzalezorlandoarm) [inline src/subcommands/create_rsa_key.rs:31]: Perhaps we should change the comment here to state that this is for PSS? Kind of obvious, but just from the comment one can not discern what is the difference between -s and -p

**Comment by tgonzalezorlandoarm** [comment]
I think `cargo fmt` should be applied here for the CI to pass

File: src/subcommands/create_csr.rs (hunk 1 above)
Comment (gowthamsk-arm) [reply src/subcommands/create_csr.rs:186]: This was just not implemented as they didn't need it.

File: src/subcommands/create_rsa_key.rs (hunk 2 above)
Comment (gowthamsk-arm) [reply src/subcommands/create_rsa_key.rs:31]: Yes, this is not updated. Will make the change.

File: src/subcommands/create_rsa_key.rs [hunk 3]
```diff
@@ -27,6 +27,10 @@ pub struct CreateRsaKey {
     #[structopt(short = 's', long = "for-signing")]
     is_for_signing: bool,
 
+    /// Supply this flag to create a signing key with PSS scheme and SHA-256 hash algorithm.
```
Comment (ionut-arm) [inline src/subcommands/create_rsa_key.rs:30]: You should probably make clear both in the code documentation and in the CLI documentation what the "priority" between these two flags is (this one and the previous).

File: tests/parsec-cli-tests.sh [hunk 4]
```diff
@@ -45,11 +45,13 @@ delete_key() {
 create_key() {
 # $1 - key type ("RSA" or "ECC")
 # $2 - key name
-# $3 - key usage ("SIGN" or "OAEP"), only consulted if $1 == "RSA"
+# $3 - key usage ("PKCS1_V15", "PSS" or "OAEP"), only consulted if $1 == "RSA"
```
Comment (ionut-arm) [inline tests/parsec-cli-tests.sh:48]: Be advised, [PKCS1 v1.5](https://en.wikipedia.org/wiki/PKCS_1#Schemes) is a name for both an encryption and for a signature scheme, so you'd need to disambiguate somewhere.

**Review by ionut-arm — COMMENTED** [review]
Thanks for the patch!

File: tests/parsec-cli-tests.sh (hunk 4 above)
Comment (gowthamsk-arm) [reply tests/parsec-cli-tests.sh:48]: Shall I rename it to `SIGN_PKCS1_V15` and `SIGN_PSS`?

Comment (ionut-arm) [reply tests/parsec-cli-tests.sh:48]: That works!

File: src/subcommands/create_rsa_key.rs (hunk 3 above)
Comment (gowthamsk-arm) [reply src/subcommands/create_rsa_key.rs:31]: Have added comments in the code and also in the CLI about this.

**Review by tgonzalezorlandoarm — APPROVED** [review]

**Review by ionut-arm — APPROVED** [review]
💯

**Pull request merged**